Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java

```diff
@@ -33,62 +39,24 @@
 final class Network implements com.breadwallet.crypto.Network {
 
     /* package */
-    static Network create(String uids, String name, boolean isMainnet, Currency currency, UnsignedLong height,
-                          Map<Currency, NetworkAssociation> associations,
-                          List<NetworkFee> fees, UnsignedInteger confirmationsUntilFinal) {
-        BRCryptoNetwork core;
-
-        String code = currency.getCode();
-        switch (code) {
-            case com.breadwallet.crypto.Currency.CODE_AS_BTC:
-                core = BRCryptoNetwork.createAsBtc(uids, name, isMainnet);
-
-                break;
-            case com.breadwallet.crypto.Currency.CODE_AS_BCH:
-                core = BRCryptoNetwork.createAsBch(uids, name, isMainnet);
-
-                break;
-            case com.breadwallet.crypto.Currency.CODE_AS_ETH:
-                Optional<BRCryptoNetwork> optional = BRCryptoNetwork.createAsEth(uids, name, isMainnet);
-                if (optional.isPresent()) {
-                    core = optional.get();
-                } else {
-                    throw new IllegalArgumentException("Unsupported ETH network");
-                }
-
-                break;
-            default:
-                core = BRCryptoNetwork.createAsGen(uids, name, currency.getCoreBRCryptoCurrency(), isMainnet);
-                break;
+    static Network findBuiltin (String uids) {
```

Comment:
Purely if you are interested; for functions that can return null, I use the Optional type.

Reply:
Updated.